src/runtime-rs/crates/persist/src/lib.rs

```diff
@@ -29,6 +29,21 @@ pub fn to_disk<T: serde::Serialize>(value: &T, sid: &str) -> Result<()> {
     Err(anyhow!("invalid sid {}", sid))
 }
 
+pub fn to_disk_jailed<T: serde::Serialize>(value: &T, sid: &str, prefix: &str) -> Result<()> {
```

Comment:
Maybe it's better to abstract `to_disk` and `to_disk_jailed` into one function, it seems there are some duplicate codes for these two functions.

Reply:
thanks @Tim-0731-Hzt! We would have been able to ditch this change altogether if that tiny bit of verify_id was a bit different (we could easily allow the '/' char by passing the prefix+'/'+sid to this function things would work as expected). 

However this is a combination of the FC jailer hardcoded jailer path (`<>/firecracker/<>`) and the fact that we didn't want to change the signature for the `to_disk` function. We'll refactor this logic and iterate over this to make sure we don't have any duplicate code in there. 
thanks!

Reply:
we have added a new parameter in the `to_disk` to account for this. However, we'd like some feedback about that, as we're not sure this captures the correct functionality for dragonball, and/or other hypervisors...